### PR TITLE
NOISSUE - Switch to Consumers interface

### LINKED ISSUE #1311: Add Notifications service
**FEATURE REQUEST**

1. Is there an open issue addressing this request? If it does, please add a "+1" reaction to the
   existing issue, otherwise proceed to step 2.
No.

2. Describe the feature you are requesting, as well as the possible use case(s) for it.
I want to create a Mainflux Writer that can send me an email when a message to some topic (channel + subtopic) is received. In the ideal case, these messages are processed using some sort of rules engine so that I can specify rules when receiving emails to avoid spam messages and unnecessary traffic, but creating an SMTP writer is the first step in that direction. Potential,  use cases are limitless, but most of them are related to notifications and alarms.

3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).
This is a should-have.

### PR DESCRIPTION
**Pull request by dborovcanin**
### What does this do?
Replace Writer interface with Consumer. 

### Which issue(s) does this PR fix/relate to?
There is no such issue.

### List any changes that modify/break current functionality
There are no such changes. This pull request will be followed by pull requests that resolve absmach/supermq#1311 and absmach/magistrala#13, which will be just another Mainflux Consumer alongside Mainflux Writers.

### Have you included tests for your changes?
No.

### Did you document any new/modified functionality?
No.

### CODE REVIEW

File: consumers/writers/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Writers
+
+Writers provide an implementation of various `message writers`.
```
Comment (drasko) [inline consumers/writers/README.md]: Is this now rather implementation of `Consumer` interface?

**Review by drasko — CHANGES_REQUESTED** [review]

File: consumers/writers/README.md (hunk 1 above)
Comment (manuio) [reply consumers/writers/README.md]: I would add a README for consumers and keep this one

File: consumers/messages.go [hunk 2]
```diff
@@ -21,42 +21,36 @@ var (
...
-
+func Start(sub messaging.Subscriber, consumer Consumer, transformer transformers.Transformer, subjectsCfgPath string, logger logger.Logger) error {
 	subjects, err := loadSubjectsConfig(subjectsCfgPath)
 	if err != nil {
 		logger.Warn(fmt.Sprintf("Failed to load subjects: %s", err))
 	}
 
 	for _, subject := range subjects {
-		if err := sub.Subscribe(subject, c.handler); err != nil {
+		if err := sub.Subscribe(subject, handle(transformer, consumer)); err != nil {
```
[the remark this answers is not in the document]
Comment (drasko) [reply consumers/messages.go]: Looking and Go closure naming [here](https://golang.org/doc/effective_go.html), this seems as an appropriate naming. Some other examples can be found [here](https://gobyexample.com/closures) or [here](https://tour.golang.org/moretypes/25).

[the remark this answers is not in the document]
Comment (darkodraskovic) [reply consumers/messages.go]: According to Effective Go, `handler()` would be a better name. Btw, the fact that the created function closes over factory parameters seems irrelevant to me here. The important part is that it's a factory function.

[the remark this answers is not in the document]
Comment (drasko) [reply consumers/messages.go]: From what I see in Effective Go, closure example is called `Announce()`, not `Announcer()`.

[the remark this answers is not in the document]
Comment (darkodraskovic) [reply consumers/messages.go]: `Announce()` there is not a factory function. It does not return a new function. It does what it says, i.e. announces things. On the contrary, our function is supposed to return us a `handler` function that is supposed to `handle`. Anyway, just a suggestion for an improved readability. I was inspired by this https://golang.org/doc/effective_go.html#Getters

**Review by manuio — APPROVED** [review]
LGTM

**Comment by darkodraskovic** [comment]
LGTM

**Review by darkodraskovic — APPROVED** [review]
LGTM

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**